.github/workflows/release.yaml

```
@@ -12,15 +13,17 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@master
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
       - name: Setup Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v3
         with:
           go-version: ${{ env.GOLANG_VERSION }}
       - name: GoReleaser
-        uses: goreleaser/goreleaser-action@v1
+        uses: goreleaser/goreleaser-action@v4
         with:
           version: latest
-          args: release --rm-dist
+          args: release --clean
         env:
-          GITHUB_TOKEN: ${{ secrets.REPO_TOKEN }}
+          GITHUB_TOKEN: ${{ github.token }}
```

```suggestion
          GITHUB_TOKEN: ${{ secrets.REPO_TOKEN }}
```

This is pulled from the secrets setup on the repo. Does `github.` work? 
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-of-onworkflow_callsecrets

---

yessir, it works. see the links to the published artifact on my fork of the repo.
the repo owner can manage the token and its permissions themselves bringing the token in through a secret as u were. u can also use the repository’s settings to control the permissions granted to the automatically provisioned token, referenced with `{{ github.token }}`. the default permissions granted to the automatically provisioned token are rather generous including the ability to publish packages. I changed it so that forks would likely build without further review though I acknowledge that the actions are now perhaps overly generously permissioned.